framework/common/src/main/java/edu/kit/kastel/mcse/ardoco/core/api/models/ModelConnector.java

```diff
@@ -1,12 +1,14 @@
 /* Licensed under MIT 2021-2023. */
 package edu.kit.kastel.mcse.ardoco.core.api.models;
 
+import java.io.Serializable;
+
 import org.eclipse.collections.api.list.ImmutableList;
 
 /**
  * The Interface IModelConnector defines the connection to a computational model.
  */
-public interface ModelConnector {
+public interface ModelConnector extends Serializable {
```

Comment:
This has the advantage that changes to the codebase invalidate persistent datarepository caches (since java automatically changes serialversionuid if major changes occur), which reduces unexpected behaviour due to caching. It has the disadvantage of requiring some memory and some performance for serialization. However, both is rather negligible compared to the effort required for serializing things such as the words, sentences and phrases. It is a tradeoff in that sense. If required, I will change it and mark the required fields transient, it is up to you